### PR TITLE
Clarified PRR freeze and exceptions.yaml in enhancements handbook

### PR DESCRIPTION
**Pull request by sreeram-venkitesh**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

This PR fixes two issues from https://github.com/kubernetes/sig-release/issues/2375

- Clarifying what PRR freeze in the enhancements handbook and updating the communication template for the enhancements freeze.
- Adding instruction to create the exceptions.yaml file after the release, along with an example.

#### Which issue(s) this PR fixes:

Fixes part of #2375

#### Special notes for your reviewer:

CC: @Priyankasaggu11929 @npolshakova

### CODE REVIEW

File: release-team/role-handbooks/enhancements/README.md
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:180]: :100:, pretty useful!

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]

**Comment by Priyankasaggu11929** [comment]
@sreeram-venkitesh, outside of this PR's scope, but a note:

we should add the blurb about PRR freeze in this document too - https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md

**Comment by Priyankasaggu11929** [comment]
cc: @salehsedghpour

File: releases/release_phases.md
Comment (Priyankasaggu11929) [inline releases/release_phases.md:5]: minor typo

```suggestion
The PRR freeze is a preliminary soft deadline, happening one week before the Enhancements Freeze.
```

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]

**Comment by Priyankasaggu11929** [comment]
@sreeram-venkitesh, could you please squash the commits also to two (one for each file change). Thanks!

**Comment by sreeram-venkitesh** [comment]
@Priyankasaggu11929 I've squashed the commits into two, one for each file! PTAL!

File: release-team/role-handbooks/enhancements/README.md
Comment (npolshakova) [inline release-team/role-handbooks/enhancements/README.md:231]: [nit[ 
```suggestion
The PRR Freeze is a preliminary soft deadline, happening a week before the [Enhancements Freeze](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md#enhancements-freeze).
```

File: release-team/role-handbooks/enhancements/README.md
Comment (npolshakova) [inline release-team/role-handbooks/enhancements/README.md:235]: Can we make the list items? 


As described by the PRR team [here](https://groups.google.com/a/kubernetes.io/g/dev/c/CQ33yPqp-H4/m/hHO-NaQiAQAJ):

* By the PRR freeze date, KEP authors must have completed the PRR questionnaire for all opted-in enhancements (KEPs).
* It is important to note that the PRR freeze **_does not_** mean that the KEPs need to have received an approval or even a review from the PRR team by this date.
* To emphasize, the sole requirement is that all opted-in KEPs have their PRR questionnaires answered by the deadline, to ensure the PRR team has sufficient time to review them by Enhancements Freeze.

**Comment by sreeram-venkitesh** [comment]
@npolshakova I've made the changes, PTAL!

File: release-team/role-handbooks/enhancements/README.md
Comment (prianna) [inline release-team/role-handbooks/enhancements/README.md:342]: Can we add some whitespace here so that markdown starts this sentence on the next line? Otherwise the template renders really wide with a horizontal scroll in preview mode.

Comment (prianna) [reply release-team/role-handbooks/enhancements/README.md:342]: Also, should we add {{ PRR_FREEZE_DATETIME }} here as well?

File: release-team/role-handbooks/enhancements/README.md
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:340]: @sreeram-venkitesh, could you add this change as well? Thanks!

During the enhancements freeze outreach, marking a KEP **_targeting_** stable as `implemented` is incorrect.

(Consider the scenario where a KEP slips to the next cycle, but kep.yaml already says `implemented`, which is misleading.)

```suggestion
- [ ] KEP status is marked as `implementable` for `latest-milestone: { CURRENT_RELEASE }`.
```

---

Should we move this line to the code freeze outreach template?

```
KEPs targeting `stable` need to be marked as `implemented` after code PRs are merged and the feature gates are removed.
```

If all PRs for stable graduation are merged, it makes sense at that point to mark the KEP as `implemented`.

However, a note: we won't push a KEP to raise a code freeze exception request if, on code freeze day, the only thing missing is marking the KEP as `implemented` in kep.yaml.

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]

File: release-team/role-handbooks/enhancements/README.md
Comment (sreeram-venkitesh) [reply release-team/role-handbooks/enhancements/README.md:340]: I agree, I was talking with @ArkaSaha30 yesterday about how it only makes sense to update the status as implemented after the code freeze. This is also an item in week 0 tasks, to contact KEPs from the previous milestone to make sure stable KEPs have this updated properly. I'll make this change.

Comment (sreeram-venkitesh) [reply release-team/role-handbooks/enhancements/README.md:340]: I've made this change, but I have a question about not pushing folks to raise code freeze exception. If a KEP has all its code PRs merged, we can mark it as tracked, even if its not marked as implemented right? Since this change can be made to k/enhancements even after the code freeze. So updating the status to `implemented` is not a criteria for passing code freeze, its just that we will remind folks to do so with the code freeze template. I think then it is better to have this in the message where we mark the KEP as tracked for code freeze. Since KEPs which don't make the code freeze cut cannot be marked as `implemented`.

Comment (sreeram-venkitesh) [reply release-team/role-handbooks/enhancements/README.md]: Please check https://github.com/kubernetes/sig-release/pull/2518/commits/38ca7c7bf71031d9e9c577abfca4ddfd53348ed0.

**Comment by Priyankasaggu11929** [comment]
@sreeram-venkitesh, just checking if it is ready for another round of review? :slightly_smiling_face:

**Comment by sreeram-venkitesh** [comment]
@Priyankasaggu11929 Yes please!

File: release-team/role-handbooks/enhancements/README.md
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:142]: How about moving the following to a line on its own & adding some annotations like "IMPORTANT TO NOTE" or similar to catch eyes!

```
Make sure you remove the `lead-opted-in` label from all the KEPs of the previous release before enabling the job. Otherwise KEPs from the previous cycle will be populated in the board.
```

File: release-team/role-handbooks/enhancements/README.md
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:148]: Actually, let's move this item on top of the instruction to update the test-infra job which updates the automation for board sync?

**Comment by Priyankasaggu11929** [comment]
@tjons @ArkaSaha30 @dipesh-rawat @mickeyboxell @prianna

could you all please give a round of review, so we get it merged max by end of this week? Thanks!

File: release-team/role-handbooks/enhancements/README.md
Comment (sreeram-venkitesh) [reply release-team/role-handbooks/enhancements/README.md:231]: We just ping them as part of our outreach and remove the KEP from the milestone if they don't make the cut. There is some overlap in what we as the enhancements team require KEP authors to complete and what the PRR team asks the KEP authors to complete. These are things like completing the questionnaire and making sure that all the questions in the KEP template are present as such. We usually just remove the KEP from the milestone during the enhancements freeze party if these are not completed by then. 

I will add a note here in the handbook adding this context and mentioning that we can remove the KEP from the milestone if this isn't completed by the enhancements freeze deadline.

**Comment by sreeram-venkitesh** [comment]
@Priyankasaggu11929 @tjons Please review the new changes! I want to get this merged before the next cycle starts!

File: release-team/role-handbooks/enhancements/README.md
Comment (Priyankasaggu11929) [reply release-team/role-handbooks/enhancements/README.md]: Hi @sreeram-venkitesh, circling back on this one

image


Let's move the first "Make sure you remove the `lead-opted-in` ..." instruction to a separate line on its own & use some emphasis (bold/italicize/quotes) to catch the eye.

**Comment by Priyankasaggu11929** [comment]
Other than my comment here https://github.com/kubernetes/sig-release/pull/2518#discussion_r1732215110, rest LGTM!

Thank you so much again for all the work, @sreeram-venkitesh.

File: release-team/role-handbooks/enhancements/README.md
Comment (sreeram-venkitesh) [reply release-team/role-handbooks/enhancements/README.md]: How does this look? I realized the warning is repeated twice, but I thought of keeping it for good measure.

image

**Comment by Priyankasaggu11929** [comment]
/lgtm
/approve

/hold
unhold if ready to merge!

Thanks again, @sreeram-venkitesh :slightly_smiling_face:

**Comment by sreeram-venkitesh** [comment]
Thanks a lot! Will unhold once @tjons also confirms the changes from his end :)

**Review by tjons — APPROVED** [review]
LGTM!

**Pull request merged**
